frontend/src/components/CardBack.tsx

```diff
@@ -329,7 +414,37 @@ export const CardBack = ({
   currentStep,
   onStepChange,
 }: CardBackProps) => {
-  const { colors } = useTheme();
+  const themeCtx = useTheme() as any;
+  const colors = themeCtx.colors;
+  const theme: 'light' | 'dark' =
+    themeCtx.theme ??
+    (colors?.background &&
+    typeof colors.background === 'string' &&
+    colors.background.toLowerCase() === '#ffffff'
+      ? 'light'
+      : 'dark');
+
+  const accent =
+    theme === 'light'
+      ? colors.primary || '#5ba20f'
+      : colors.primary || PRIMARY_ACCENT_COLOR;
+
```

Comment:
This is too messy, we should hide all of this style/theme specific stuff inside the useTheme hook